dynamics/client.py

```diff
@@ -341,6 +358,7 @@ def get(self, *, not_found_ok: bool = False, query: Optional[str] = None) -> Lis
             url=query,
             headers={**self.default_headers("get"), **self.headers},
         )
+        response.raise_for_status()
```

Comment:
Raising early here will bypass the DynamicsClient's built-in error handler ([handled_error](https://github.com/MrThearMan/dynamics-client/pull/4/files#diff-fafb0b7303b9394168b2c6f6aef7be16bcc89b26745d80f08eaa4f03eb37bb5bR321)), which coerces the errors from the http client to the errors specified in the [error_dict](https://github.com/MrThearMan/dynamics-client/pull/4/files#diff-fafb0b7303b9394168b2c6f6aef7be16bcc89b26745d80f08eaa4f03eb37bb5bR71). This allows users to define the errors they can get from the client by subclassing, and also automatically raises `rest_framework.APIException` when using [drf](https://www.django-rest-framework.org/). We also get nicer error messages from the [error-key](https://github.com/MrThearMan/dynamics-client/pull/4/files#diff-fafb0b7303b9394168b2c6f6aef7be16bcc89b26745d80f08eaa4f03eb37bb5bR375) in the response data, when that is available (although not in this case).

I would like to keep this behavior, and handle the special case where we can't parse the respose as json for the error message separately. I think the [handle_error here](https://github.com/MrThearMan/dynamics-client/pull/4/files#diff-fafb0b7303b9394168b2c6f6aef7be16bcc89b26745d80f08eaa4f03eb37bb5bR366) already does this (if I'm not wrong), although it could return the actual `response.status_code` instead of always 500.

Reply:
Please see the new commit and let me know if it works for you.

Reply:
Yes, adding it to the try-except block should work. I'd prefer separate try-except blocks for each error, since `response.raise_for_status()` should only raise `HTTPError` and not `JSONDecodeError` and vice verca, but it's fine in this case. 

However, you shouldn't mix the async functionality in with this PR, and make a separate PR for that.